daemon/commit.go

```diff
@@ -228,7 +228,7 @@ func (daemon *Daemon) Commit(name string, c *backend.ContainerCommitConfig) (str
 
 	imageRef := ""
 	if c.Repo != "" {
-		newTag, err := reference.WithName(c.Repo) // todo: should move this to API layer
+		newTag, err := reference.ParseNormalizedNamed(c.Repo) // todo: should move this to API layer
```

Comment:
Not sure if it's possible for `c.Repo` to contain `:` or `@` characters, but it might be good to be defensive and return an error if `!IsNameOnly(newTag)`